src/Server.cpp

```diff
@@ -656,6 +626,12 @@ void cServer::KickUser(int a_ClientID, const AString & a_Reason)
 
 void cServer::AuthenticateUser(int a_ClientID, const AString & a_Name, const AString & a_UUID, const Json::Value & a_Properties)
 {
+	if (GetNumPlayers() >= GetMaxPlayers())
```

Comment:
I was thinking if two threads incremented the count simultaneously and then proceeded to check the condition if would fail for both, but in fact, surely this function only ever is called by one thread per server run, since either authentication is on (and the single auth. thread calls it) or authentication is off, in which the server thread calls it.

Then if this is right, synchronisation is not needed...?